### PR TITLE
improved logging facility

### PR DESCRIPTION
**Pull request by jedwards4b**
The logging facility now includes a -s, --silent option.  
When a script is run at the --debug level it also creates a file scriptname.log

So there are now 4 levels of logging:
default (no option)
--silent
--verbose
--debug

### CODE REVIEW

File: scripts/Tools/standard_script_setup.py [hunk 1]
```diff
@@ -15,5 +16,5 @@
 import CIME.utils
 CIME.utils.check_minimum_python_version(2, 7)
 CIME.utils.stop_buffering_output()
-
-logging.basicConfig()
+# set up logging to file
+logging.basicConfig(level=logging.INFO)
```
Comment (jgfouca) [inline scripts/Tools/standard_script_setup.py]: Doesn't this turn on --verbose always?

File: utils/python/CIME/XML/component.py [hunk 2]
```diff
@@ -6,6 +6,8 @@
 from entry_id import EntryID
 from CIME.utils import expect, get_cime_root, get_model
 
+logger = logging.getLogger(__name__)
```
Comment (jgfouca) [inline utils/python/CIME/XML/component.py]: Does this mean each .py file gets its own logger?

File: utils/python/CIME/system_test.py [hunk 3]
```diff
@@ -606,12 +608,12 @@ def _consumer(self, test, test_phase, phase_method):
                      (test_phase, test, elapsed_time, status)
         if not success:
             status_str += "    Case dir: %s\n" % self._get_test_dir(test)
-        sys.stdout.write(status_str)
+        logger.info(status_str)
 
         # On batch systems, we want to immediately submit to the queue, because
         # it's very cheap to submit and will get us a better spot in line
         if (success and not self._no_run and not self._no_batch and test_phase == MODEL_BUILD_PHASE):
-            sys.stdout.write("Starting %s for test %s with %d procs\n" % (RUN_PHASE, test, 1))
```
Comment (jgfouca) [inline utils/python/CIME/system_test.py]: Not sure I agree with this. Without --verbose, create_test won't display any info. Also, the sys.stdout.write was necessary for threaded printing. Don't know if logger is thread safe.

File: scripts/Tools/standard_script_setup.py (hunk 1 above)
Comment (jedwards4b) [inline scripts/Tools/standard_script_setup.py]: I moved some stuff that we had set to info into debug, the idea being that we want the default output level to be more informational than warn.

File: utils/python/CIME/XML/component.py (hunk 2 above)
Comment (jedwards4b) [inline utils/python/CIME/XML/component.py]: yes they do, but by default all loggers go to the same handler.  try running with the different levels to see how each is treated.

File: utils/python/CIME/utils.py [hunk 4]
```diff
@@ -149,7 +150,7 @@ def run_cmd(cmd, ok_to_fail=False, input_str=None, from_dir=None, verbose=None,
     if (arg_stderr is _hack):
         arg_stderr = subprocess.PIPE
 
-    logging.info("RUN: %s" % cmd)
```
Comment (jgfouca) [inline utils/python/CIME/utils.py]: This does not take the verbose flag to this function into account.

File: utils/python/CIME/system_test.py (hunk 3 above)
Comment (jedwards4b) [inline utils/python/CIME/system_test.py]: logger is thread safe and in fact at debug level will tell you which thread printed each message.

File: scripts/Tools/standard_script_setup.py (hunk 1 above)
Comment (jgfouca) [inline scripts/Tools/standard_script_setup.py]: OK, so logging.info(...) is something users will always see?

Comment (jedwards4b) [inline scripts/Tools/standard_script_setup.py]: By default, yes.   --silent turns it off.

File: utils/python/CIME/utils.py (hunk 4 above)
Comment (jedwards4b) [inline utils/python/CIME/utils.py]: I don't understand this comment?

Comment (jgfouca) [inline utils/python/CIME/utils.py]: If you look above, this function takes a verbose flag. The idea is, some shell commands are so important that you always want to see "verbose" information about them. In most cases, the shell commands are not interesting and you don't want to see anything with default logging.

**Comment by jgfouca** [comment]
Closed, made new PR.

**Pull request merged**